projects/swimlane/ngx-ui/src/lib/components/dialog/dialog.component.ts

```diff
@@ -151,15 +161,30 @@ export class DialogComponent implements OnInit, OnDestroy {
     return this.closeOnBlur && target.classList.contains('dialog');
   }
 
-  @HostListener('keydown.esc')
-  onEscapeKeyDown(): void {
-    if (this.closeOnEscape) this.hide();
-  }
-
-  @HostListener('document:click', ['$event.target'])
-  onDocumentClick(target: any): void {
-    if (this.containsTarget(target)) {
-      this.hide();
-    }
+  private setupEventListeners(): void {
+    this.ngZone.runOutsideAngular(() => {
+      const keydownEscListener = this.renderer.listen(this.element.nativeElement, 'keydown', (event: KeyboardEvent) => {
+        // We could've actually used `render.listener(element, 'keydown.esc')` and that would definitely work.
+        // Unfortunately, Angular delegates the event handling to the `KeyEventsPlugin` when it meets the `.` (dot)
+        // in the event name. The `KeyEventsPlugin` calls `ngZone.runGuarded` which triggers change detection anyway
+        // (no matter if we added event listeners in the root zone).
+        // In that case the change detection will be run only when the user clicks `esc` button.
+        if (event.keyCode === 27 && this.closeOnEscape) {
```

Comment:
nitpick: let's use `Keycodes.ESCAPE` from AngularCDK. When they deprecate Keycodes and switch over to Keys, we will be able to make the change here as well.

Reply:
Thank you for bringing new knowledge! Updated the code.